src/chapters/interfaces-and-polymorphism/exercises.rst

```diff
@@ -0,0 +1,202 @@
+
+   a. Import ``java.util.Comparator``. This removes the flag on ``Comparator``.
+   b. Hover over the line again and select *implement methods*. Choose the
+      ``compare`` option.
+
+      .. figure:: figures/implement-methods.png
+         :alt: Selection menu for methods to implement.
+
+   c. This adds an ``@Override`` method that compares two ``Flavor`` objects
+      and *always* returns ``0``.
+
+      .. sourcecode:: Java
+         :lineno-start: 6
+
+         @Override
+         public int compare(Flavor o1, Flavor o2) {
+            return 0;
+         }
+
+#. Always returning ``0`` results in no sorting, so replace line 8 with:
+
+   .. sourcecode:: Java
+
+      return o1.getName().compareTo(o2.getName());
+
+   This returns an integer (negative, positive, or zero) depending on
+   whether ``o1`` or ``o2`` comes first, alphabetically.
+
+Sorting the ``flavors`` ArrayList
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In ``Main``, we declare ``allItems`` that contains everything in the ``Case``
+as well as specific ``flavors`` and ``cones`` collections.
+
+.. sourcecode:: Java
+   :lineno-start: 6
+
+   public static void main(String[] args){
+      Case allItems = new Case();
+      ArrayList<Flavor> flavors = allItems.getFlavors();
+      ArrayList<Cone> cones = allItems.getCones();
+
+   }
+
+#. To sort the ``flavors`` list, first create a new ``FlavorComparator``
+   object.
+
+   .. sourcecode:: Java
+      :lineno-start: 6
+
+      public static void main(String[] args){
+         Case allItems = new Case();
+         ArrayList<Flavor> flavors = allItems.getFlavors();
+         ArrayList<Cone> cones = allItems.getCones();
+         Comparator comparator = new FlavorComparator();
+      }
+
+#. Next, call the ``sort`` method on ``flavors`` and pass the ``comparator``
+   object as the argument.
+
+   .. sourcecode:: Java
+      :lineno-start: 6
+
+      public static void main(String[] args){
+         Case allItems = new Case();
+         ArrayList<Flavor> flavors = allItems.getFlavors();
+         ArrayList<Cone> cones = allItems.getCones();
+         Comparator comparator = new FlavorComparator();
+
+         flavors.sort(comparator);
+      }
+
+#. Iterating through the ``flavors`` list before and after the sort shows
+   the results. (The output below displays just the ``name`` fields).
+
+   .. sourcecode:: bash
+
+      Before:                 After:
+
+      Vanilla                 Chocolate
+      Chocolate               Red Velvet
+      Red Velvet              Rocky Road
+      Rocky Road              Strawberry Sorbet
+      Strawberry Sorbet       Vanilla
+
+Note that ``Main`` does NOT have to implement the ``Comparator`` interface.
+This only needs to happen in the class that actually uses the ``compare``
+method.
+
+.. admonition:: Tip
+
+   Instead of declaring and initializing a ``comparator`` object, we could
+   condense lines 10 & 12 into a single statement:
```

Comment:
Lines 10 & 12 of what file? I would hesitate to refer to specific line numbers because some people really space things out.

Reply:
I rephrased the admonition. Let me know what you think.